internal/dataplane/parser/translators/ingress.go

```diff
@@ -0,0 +1,225 @@
+package translators
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/kong/go-kong/kong"
+	networkingv1 "k8s.io/api/networking/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+// -----------------------------------------------------------------------------
+// Ingress Translation - Public Functions
+// -----------------------------------------------------------------------------
+
+// TranslateIngress receives a Kubernetes ingress object and from it will
+// produce a translated set of kong.Services and kong.Routes which will come
+// wrapped in a kongstate.Service object.
+func TranslateIngress(ingress *networkingv1.Ingress) []*kongstate.Service {
+	index := &ingressTranslationIndex{cache: make(map[string]*ingressTranslationMeta)}
+	index.add(ingress)
+	kongStateServices := kongstate.Services(index.translate())
+	sort.Sort(kongStateServices)
+	return kongStateServices
+}
+
+// -----------------------------------------------------------------------------
+// Ingress Translation - Private Consts & Vars
+// -----------------------------------------------------------------------------
+
+var defaultHTTPIngressPathType = networkingv1.PathTypePrefix
+
+const (
+	defaultHTTPPort       = 80
+	defaultServiceTimeout = 60000
```

Comment:
How about we make this a typed time interval (e.g. in seconds) and then where this is used we call `.Seconds()`?

Reply:
Given that there is no functional need to change this type from int, as it's a private variable only used in one place, and only accepted by the relevant API as an int, but understanding that the reason you're asking for this change is probably just for better at-a-glance understanding of what the value is, I opted to add a godoc comment to this const in 73d64f146992cc6c0c43c2e59ce2b16884186e90. Let me know if that seems like an adequate compromise?

Reply:
The reason I suggested this is to leverage the type system, make it obvious to the reader what unit this is in and at the same time making it harder to make a mistake when this needs a change in the future.

Reply:
I take your response as meaning you don't feel this is a reasonable compromise and so in the interest of time and this patch being fairly urgent I've just opted to do it as you requested originally (albeit, calling `.Milliseconds()` specifically). This was pushed up in 39b33a0442a2d394d997fd4529261bd954957756